networkx/algorithms/operators/product.py

```diff
@@ -508,22 +508,17 @@ def corona_product(G, H):
     GH.add_nodes_from(G)
     GH.add_edges_from(G.edges)
 
-    new_nodes = []
-    new_edges = []
-
-    for G_node in GH:
+    for G_node in G:
 
         # copy nodes of H in GH, call it H_i
-        new_nodes.extend([(G_node, v) for v in H])
+        GH.add_nodes_from([(G_node, v) for v in H])
```

Comment:
You can even go one step further by removing the square brackets in the function call. That creates a "generator expression" that avoids creating the list. It just yields one node (or edge) at a time as needed.

`GH.add_nodes_from((G_node, v) for v in H)`